pkg/diagnostics/workflow_metrics.go

```diff
@@ -0,0 +1,162 @@
+)
+
+type workflowMetrics struct {
+	// Total successful workflow operation requests
+	workflowOperationsTotal *stats.Int64Measure
+	// Total failed workflow operation requests
+	workflowOperationsFailedTotal *stats.Int64Measure
+	// Workflow operation request's Response latency
+	workflowOperationsLatency *stats.Float64Measure
+
+	// Total workflow/activity reminders created
+	workflowRemindersTotal *stats.Int64Measure
+
+	// Total successful workflow/activity executions
+	workflowExecutionTotal *stats.Int64Measure
+	// Total failed workflow/activity executions
+	workflowExecutionFailedTotal *stats.Int64Measure
+	// Time taken to run a workflow to completion
+	workflowExecutionLatency *stats.Float64Measure
+
+	// Latency between execution request and actual execution
+	workflowSchedulingLatency *stats.Float64Measure
+
+	appID   string
+	enabled bool
+}
+
+func newWorkflowMetrics() *workflowMetrics {
+	return &workflowMetrics{
+		workflowOperationsTotal: stats.Int64(
+			"runtime/workflow/operations/total",
+			"The number of successful workflow operation requests.",
+			stats.UnitDimensionless),
+
+		workflowOperationsFailedTotal: stats.Int64(
+			"runtime/workflow/operations/failed_total",
+			"The number of failed workflow operations requests.",
+			stats.UnitDimensionless),
+		workflowOperationsLatency: stats.Float64(
+			"runtime/workflow/operations/latency",
+			"The latencies of responses for workflow operation requests.",
+			stats.UnitMilliseconds),
+		workflowRemindersTotal: stats.Int64(
+			"runtime/workflows/reminders/total",
+			"The number of workflows/activity reminders created.",
+			stats.UnitDimensionless),
+		workflowExecutionTotal: stats.Int64(
+			"runtime/workflow/execution/total",
+			"The number of successful workflow/activity executions.",
+			stats.UnitDimensionless),
+		workflowExecutionFailedTotal: stats.Int64(
+			"runtime/workflow/execution/failed_total",
+			"The number of failed workflow/activity executions.",
+			stats.UnitDimensionless),
+		workflowExecutionLatency: stats.Float64(
+			"runtime/workflow/execution/latency",
+			"The total time taken to run a workflow/activity to completion.",
+			stats.UnitMilliseconds),
+		workflowSchedulingLatency: stats.Float64(
+			"runtime/workflow/scheduling/latency",
+			"The latency between execution request and actual execution.",
+			stats.UnitMilliseconds),
+	}
+}
+
+func (w *workflowMetrics) IsEnabled() bool {
+	return w != nil && w.enabled
+}
+
+// Init registers the workflow metrics views.
+func (w *workflowMetrics) Init(appID string) error {
+	w.appID = appID
+	w.enabled = true
+
+	return view.Register(
+		diagUtils.NewMeasureView(w.workflowOperationsTotal, []tag.Key{appIDKey, operationKey}, view.Count()),
+		diagUtils.NewMeasureView(w.workflowOperationsFailedTotal, []tag.Key{appIDKey, operationKey}, view.Count()),
+		diagUtils.NewMeasureView(w.workflowOperationsLatency, []tag.Key{appIDKey, elapsedKey}, defaultLatencyDistribution),
```

Comment:
Similar to Count metrics above, the same Keys can be added here

Reply:
Combined into one.